climada_petals/entity/exposures/openstreetmap/osm_dataloader.py

```diff
@@ -0,0 +1,945 @@
+        file_format : str
+            Format in which file should be downloaded; options are 
+            ESRI Shapefiles (shp), which can easily be loaded into gdfs,
+            or osm-Protocolbuffer Binary Format (pbf), which is smaller in 
+            size, but has a more complicated query syntax to load (functions
+            are provided in the OSMFileQuery class).
+        save_path : str or pathlib.Path
+            Folder in which to save the file
+            
+        Returns
+        -------
+        None
+            File is downloaded and stored unter save_path + filename
+        
+        See also
+        --------
+        DICT_GEOFABRIK for exceptions / special regions.
+        """
+        
+        download_url = self._create_gf_download_url(iso3, file_format)
+        local_filepath = Path(save_path , download_url.split('/')[-1])
+        if not Path(local_filepath).is_file():
+            LOGGER.info(f'Downloading file as {local_filepath}')
+            urllib.request.urlretrieve(download_url, local_filepath)
+        else:
+            LOGGER.info(f'file already exists as {local_filepath}')
+
+    def get_data_planet(self, 
+                        save_path=Path(DATA_DIR,'planet-latest.osm.pbf')):
+        """
+        Download the entire planet file from the OSM server (ca. 60 GB).
+        
+        Parameters
+        ----------
+        save_path : str or pathlib.Path
+        """
+        
+        if not Path(save_path).is_file():
+            LOGGER.info(f'Downloading file as {save_path}')
+            urllib.request.urlretrieve(self.planet_url, save_path)
+        else:
+            LOGGER.info(f'file already exists as {save_path}')
+
+    def _osmosis_extract(self, shape, path_planet, path_extract):
+        """
+        Runs the command line tool osmosis to cut out all map info within 
+        shape, from the osm planet file, unless file already exists.
+        
+        Parameters
+        -----------
+        shape : list or str
+            bounding box [xmin, ymin, xmax, ymax] or file path to a .poly file
+        path_planet : str or pathlib.Path
+            file path to planet.osm.pbf
+        path_extract : str or pathlib.Path
+            file path (incl. name & ending) under which extract will be stored
+            
+        Returns
+        -------
+        None
+            Extracts and stores a new file as indicated under path_extract
+        """
+
+        if not Path(path_extract).is_file():
+            
+            LOGGER.info('File doesn`t yet exist. Assembling osmosis command.')
+            if (isinstance(shape, list) or isinstance(shape, tuple)):
+                cmd = ['osmosis', '--read-pbf', 'file='+str(path_planet), 
+                       '--bounding-box', f'top={shape[3]}', f'left={shape[0]}',
+                       f'bottom={shape[1]}', f'right={shape[2]}', 
+                       '--write-pbf', 'file='+str(path_extract)]
```

Comment:
Will this run on Windows? 😬 
Could be worth catching errors and suggesting common reasons it fails (you're on Windows, osmosis not installed, etc)

Reply:
I am not sure how to catch command-line tool errors.
It is supposed to run on windows, when installed 😄

Reply:
i also don't know. maybe @emanuel-schmid can tell us if we should worry.
if it's important we can make an issue